include/llvm/ADT/STLExtras.h

```diff
@@ -200,6 +211,91 @@ inline mapped_iterator<ItTy, FuncTy> map_iterator(const ItTy &I, FuncTy F) {
   return mapped_iterator<ItTy, FuncTy>(I, F);
 }
 
+/// An iterator adaptor that filters the elements of given inner iterators.
```

Comment:
I don't follow why this and the tests below were required. It doesn't look like they are used outside of the tests?

Reply:
I pulled this change in because the other changes conflicted without it. It isn’t necessary, but it made merging easier.